### ISSUE TITLE
Slack vs Email for community conversations

### ISSUE THREAD

**Issue by ErinBecker** [carpentries/conversations#6]
From a comment to the [Feedback about Communications](https://software-carpentry.org/blog/2016/12/conversations.html) SWC blog post by Meredith Rawls:

I'd be much more active in conversations in the Carpentry community if it shifted away from email and toward Slack. Slack fosters a casual communication style and avoids unwieldy threaded email replies and/or digest mailings; it is easy to catch up on missed posts or only read about topics of interest. Slack users don't feel like they're inconveniencing people by sending email to everyone and can have shorter, real-time "instant message" style communication with folks who are currently online. 
On a positive note, I do like how Carpentry makes heavy use of GitHub for collaboratively developing lessons. And there are many easy ways to integrate things happening on GitHub directly into channels on Slack!

**Comment by ErinBecker** [comment]
My initial response there:

Hi Meredith,

Thanks for your comments and sorry it took me a few days to follow up. Tracy and I have been teaching instructor training the past two days and I fell behind on checking this post.

I know that there has been some conversation in the community previously about the benefits of Slack vs. email. We have a Slack channel (swcarpentry.slack.com), but it has had very low volume and seems not to have been widely adopted by community members. I'd be very interested in hearing what others think about Slack vs email as a primary communication mechanism and also in hearing more about how we can integrate Slack and GitHub. There's also the question of whether a community forum might replace email lists. A forum has many of the benefits you mention of Slack (easy to read only topics of interest, avoids lengthy digests), but also wasn't widely adopted by the community when Data Carpentry initiated a forum some time ago (http://discuss.datacarpentry.o....

I'll be cross posting your comment to the GitHub conversations repo to see if we can get more input from the community here and there about optimal communication channels. One complication of trying to have conversations about the best way to do communications is that people already have established communications preferences - trying to make sure the whole community can engage in the conversation on their prefered channel requires some back and forth!

**Comment by wking** [comment]
On Thu, Dec 15, 2016 at 09:49:11AM -0800, Erin Becker wrote:
> I'd be very interested in hearing what others think about Slack vs
> email as a primary communication mechanism and also in hearing more
> about how we can integrate Slack and GitHub.

My main concern with Slack is the lack of public, linkable archives.
But you can work around that (e.g. with a bot that monitors Slack and
logs it to a public location).

Another issue is that Slack/Twitter and email/GitHub facilitate
different types of communication.  The response I'm writing now would
not be a good match for a Slack message ;).

> One complication of trying to have conversations about the best way
> to do communications is that people already have established
> communications preferences - trying to make sure the whole community
> can engage in the conversation on their prefered channel requires
> some back and forth!

I think this is important.  And the mitigation/solution is to pick a
channel that interoperates well with whatever folks are already using.
I'm sending this response via my mail user agent, because I like
email.  Others might read and respond via GitHub's web UI because they
like their browser.

Slack's GitHub intergration looks like it's one-way (GitHub → Slack
[1]), so folks following this issue on Slack wouldn't be able to
respond *from Slack*.  However, Slack does provide an IRC gateway [2],
so IRC users can connect to Slack from their IRC client (but I'm not
aware of a way for Slack to connect to an external IRC channel).  It
also looks like there's a (non-free) email ↔ Slack bridge [3,4] and a
Slack → email forwarder (for Slack Standard Plan and up) [5].

And the fallback to “we don't have a good bridge yet” (e.g. GitHub ↔
Disqus here) is manually cross-posting interesting ideas, summaries,
or just “there's a good discussion of Slack vs. email over in [6] if
anyone wants to join in”.

[1]: https://get.slack.help/hc/en-us/articles/232289568-Use-GitHub-with-Slack
[2]: https://get.slack.help/hc/en-us/articles/201727913-Connect-to-Slack-over-IRC-and-XMPP
[3]: https://slack.com/apps/A0JUW1X96-mailclark-the-email-bot
[4]: https://mailclark.ai/pricing
[5]: https://slack.com/apps/A0F81496D-email
[6]: https://software-carpentry.org/blog/2016/12/conversations.html

**Comment by willingc** [comment]
Have you looked at Zulip (https://www.zulip.org) which is open source? It's quite nice. @timabbott would be able to provide more insights if you would like more information.

**Comment by jduckles** [comment]
@willingc I have looked at Rocket Chat (https://rocket.chat/) but I'm reluctant to run more critical infrastructure on our own until we can bring on a part-time system administrator.

**Comment by timabbott** [comment]
Howdy!  Yeah, Zulip is 100% open source, with a large development community (our last release had 3000 commits from 140 people).  It's designed around a unique threading model which makes it easy and efficient for people to catch up on exactly the conversations that are important to them, which is especially great for part-time communities where keeping up can be a significant portion of someone's total available time.  You can check out how it works in the Zulip development community by signing up on chat.zulip.org (notes on how that community works here: http://zulip.readthedocs.io/en/latest/chat-zulip-org.html).  https://www.recurse.com/blog/112-how-rc-uses-zulip may also be useful context.

There's a beta (production from a technical perspective, but not yet launched since landing pages aren't ready) Zulip-powered SaaS service that can provide a Zulip instance for Software Carpentry free of charge, if running infrastructure is a concern; so you don't need a sysadmin to use Zulip.

**Comment by rgaiacs** [comment]
@Cadair and @dpshelio told me that https://riot.im/ worked well (so far) for Open Astronomy.

**Comment by Cadair** [comment]
I strongly recommend using Riot/matrix as it does the following things which I think are really valuable for an open source community:

* Can be configured to not require any user account.
* Open source protocol and clients.
* Freely hosted by matrix.org /riot.im
* Can be bridged to IRC, Slack and Gitter (hosted by Riot) so people can use whatever platform they are already tied into.
* Good quality mobile applications for iOS and Android.

More generally I think the matrix protocol is the best solution to the problem of all these web chat services that we all end up using, as it supports federation and is fully open source.

**Comment by wking** [comment]
On Thu, Mar 02, 2017 at 03:15:15AM -0800, Stuart Mumford wrote:
> More generally I think the matrix protocol is the best solution to
> the problem of all these web chat services that we all end up using,
> as it supports federation and is fully open source.

There are many open-source chat services and protocols (e.g. IRC has
an RFC [1]).  The great thing about Matrix is that it's decentralized
(so no single point of failure) and theres's a walkable directed
acyclic graph of events [2] (so you can easily catch back up after
being offline for a time [3]).  That means archives are a first-class
citizen, and you can migrate the discussion (and all the history) over
to a new location if the original hosts decide to stop hosting the
discussion.

I've setup #carpentries:matrix.org if folks are interested in a Matrix
channel, and we can setup bridges to any other Carpentry services
(e.g. the Slack channel?) if other folks show up on Matrix and whoever
admins the Slack channel is interested ;).

[1]: https://tools.ietf.org/html/rfc1459
[2]: https://matrix.org/docs/spec/intro.html#event-graphs
[3]: https://matrix.org/docs/spec/server_server/unstable.html#backfilling
[4]: http://matrix.org/docs/projects/try-matrix-now.html#application-services

**Comment by Cadair** [comment]
I don't seem to be able to join #carpentries:matrix.org, I get "room
does not exist".

**Comment by wking** [comment]
> I don't seem to be able to join #carpentries:matrix.org, I get "room
does not exist".

Forot to make it public ;).  Should be good now.

**Comment by dpshelio** [comment]
I don't believe one tool (slack/matrix/...) is a substitute of email - even though I've run [an unconference against email](http://sciencetogether.online/). I like to have a "chatting" tool to discuss things in a quick way and talk about stuff that may not be of large importance, getting quick replies on whatever I ask (*e.g.*, something is not installing well in a workshop). However, I would still prefer to use email in the way it's used at the moment in the carpentries: as a forum. I won't be able to scroll 1000 messages up to see what the people has discussed while I was offline, but I can quickly skim through lots of email subjects to see what I'm interested. 
Of course, if email is not used "the right way" (if there's such) then conversations could be diluted and hidden on different threads.

**Comment by tracykteal** [comment]
Thanks, good point. It does seem like there are different types of communication that have different tool needs. One of the biggest challenges with any chat tool is having enough people hanging out on the channel for there to be conversations or be able to get answers to quick questions.

**Comment by rgaiacs** [comment]
I received a few emails saying

> An organization owner has approved the following application to access private data in the Software Carpentry organization:
>
>   "Gitter Public Repo Access" from Gitter
>
> Because you are a member of the Software Carpentry organization and have already granted this application access to your personal account, Gitter Public Repo Access can now access Software Carpentry organization resources on your behalf.

Because I didn't see any announcement on [Software Carpentry website](github.com/swcarpentry/website) neither on the [mailing list](http://lists.software-carpentry.org/pipermail/discuss/) I though would be appropriate to ask here what the plans are because I don't think that keep [Slack](https://swcarpentry.slack.com/) and [Gitter](https://gitter.im/swcarpentry) running at the same time will be beneficial to us.

@ahmadia The Software Carpentry Gitter welcome page has your name on it so I think you could be the right person to provide some explanation to me. :-)

cc @weaverbel @ErinBecker @jduckles

**Comment by tracykteal** [comment]
I also got those notifications, but am not sure what they're about.

**Comment by Cadair** [comment]
Having initially commented on this issue talking about alternatives to slack, I would like to make the comment that any real-time communication method should not be considered a replacement for email, or similar asynchronous communication.

A good post on the advantages of email-like communication can be found here: https://dave.cheney.net/2017/04/11/why-slack-is-inappropriate-for-open-source-communications

**Comment by bradfordcondon** [comment]
Just joining the organization as a big fan of Slack (while acknowledging the drawbacks brought up in this thread).

If Slack communication is at a point where its worth sharing with the broader community, maybe a digest of some sort would be a good idea. either manually or [automated](https://slackdigest.com/)?
